### DISCUSSION TITLE
O3D303, unable to successfully update firmware, now not reachable via iVA nor ifm3d

### DISCUSSION THREAD

**Discussion by WILL-VESTERGAARD** [ifm/ifm3d#443] [Q&A]
I'm new to this, so bare with me!
Updating the firmware on my 03D303 **seems** to go ahead fine, but I am unable to verify the outcome (nor use this device at all now) because:

- Using ifm3d (CLI) and `swupdate`, the firmware update proceeds before returning `Timed out waiting for producitve mode`, (#416), but after returning to productive mode, any attempt to reach the camera returns the following:

  `←[91m2024-02-16 23:22:54 ERROR [C:\builds\syntron\support\csr\ifm3d\ifm3d\modules\device\src\libifm3d_device\xmlrpc_wrapper.hpp:100] http://192.168.0.69:80/api/rpc/v1/com.ifm.efector/ -> getParameter: RPC response indicates failure.  XMLRPC object not found (D-Bus error 22)←[0m←[0K`
  
  `←[91m2024-02-16 23:22:54 ERROR [C:\builds\syntron\support\csr\ifm3d\ifm3d\modules\device\src\libifm3d_device\device.cpp:184] While trying to instantiate camera: Sensor: XMLRPC obj not found - trying to access dead session?: XMLRPC object not found (D-Bus error 22)←[0m←[0K`
  
  `ifm3d error: 100000`
  `Sensor: XMLRPC obj not found - trying to access dead session?: XMLRPC object not found (D-Bus error 22)`

- Using the web interface at `(http://192.168.0.69:8080/)`, similarly, the firmware update begins, but stalls at _Waiting for requests.._. See the following:

  `Waiting for requests...`
  `SWUPDATE successful !`
  `Installing image zImage-goldeneye.itb into volume /dev/ubi0_0(fitimage)`
  `Installing image vantage-image-goldeneye.ubifs into volume /dev/ubi1_0(rootfs)`
  `Installing image ifm-ge-apps_1.71.6_armv7a-vfp-neon.ubifs into volume /dev/ubi1_1(ifm)`
  `DIAGCTL update SUCCESS!`
  `Installation takes a while ...`
  `DIAGCTL File: merida-v...bin`
  `Installation in progress`
  `Our hardware: O3D303 A300-D322-F342-I300-M381`
  `Software Update started !`
  `File information: O3D3XX_Firmware_1.71.9079.swu size: 139992576 bytes`

- Using iVA, the firmware update proceeds once again, before stalling and then finally returning an update failure message due to:
 `System reboot failed. 120004` 

- Rebooting the device to productive mode in iVA (after any of the above update methods) causes a stall once again while booting productive mode before returning a reboot failure with a familiar reason:
`System reboot failed. 120004` 

- Searching for the device in iVA when in productive mode shows it listed, but connecting stalls before returning a device detection failure due to:
 `Could not find a compatible device at the given IP address. 120015`

I'm surely doing something wrong, so any help would be greatly appreciated, my expertise in this area is quite limited!

**Comment by desengph** [comment]
Hello @WILL-VESTERGAARD we are looking into this and will let you know as soon as possible once we have a solution available.

Can you please let us know what exact versions you are working on:
+ ifm3d API version
+ O3D FW version before the update
+ O3D FW version to be installed
+ iVA version

**Reply by WILL-VESTERGAARD** [reply]
Hi,

Thank you for your reply!

- ifm3d API version
Attempted: 1.0.1, 1.5.1, same outcome, (other than less verbose CLI output on 1.0.1).

- O3D FW version before the update
Not known.

- O3D FW version to be installed
Attempted: 1.71.9079, 1.71.7444.

- iVA version
Attempted: 2.2.11, 2.7.6.

Thanks again!

**Reply by graugans** [reply]
From the log above

```
Waiting for requests...
SWUPDATE successful !
...
```

It looks like you already have updated the device, what is missing is to click on `Reboot to productive mode`...

**Reply by graugans** [reply]
If this does not help, and the device comes up in recovery again. You might have to send it back to ifm.
